src/tests/ftest/server/metadata.py

```diff
@@ -357,3 +357,73 @@ def test_metadata_server_restart(self):
 
                 # Start the servers
                 self.start_server_managers()
+
+    @avocado.fail_on(DaosApiError)
+    def test_container_removal_after_der_nospace(self):
+        """JIRA ID: DAOS-4858.
+
+        Test Description:
+           Verify container can be successfully deleted when the storage pool
+           is full ACL grant/remove modification.
+
+        :avocado: tags=metadata,metadata_der_nospace,nvme,large,hw
+        :avocado: tags=full_regression,der_nospace
+        """
+        self.pool.pool.connect(2)
+        init_container = NO_OF_MAX_CONTAINER
+        additional_container = 100
```

Comment:
I'm curious how you picked this number. i.e., how many additional containers can you create after creating 4150 containers?

Reply:
From the test log, der_nospace (RC: -1007) triggered on the 41th additional containers, 100 would be around the middle number.

Reply:
Shouldn't these values be based on the actual storage? If we upgrade our test cluster to servers with more storage, or someone runs these tests on their own cluster with more storage, these tests won't work correctly.

Reply:
Thanks for the comment.
Current test is creating the NO_OF_MAX_CONTAINER (a global variable in the script) to trigger the RC: -1007 and make sure we able able to clean the container and create new containers.

Reply:
Hmm, ok. I looked at the code, and the comment above `NO_OF_MAX_CONTAINER` seems to indicate that it is an arbitrary number that is likely to be incorrect on systems with more or less pmem storage than our current CI nodes. But that's not your problem to fix here, and I guess as long as everything is relative to that value, it won't be too hard to make that value dynamic based on the actual storage found.

Reply:
Thanks for the suggestion.
To check a actual pool usage is a good approach, I will create a JIRA to make sure this is covered and tested in sprint 1.3.

Reply:
Hmm, how about looping until you hit -1007instead of picking a number?

Reply:
Thanks for the comment. 
If we use loop until, test script might run into infinitive loop in case the container create did not break, or have to wait until test timeout.
I increased the additional container in loop to a limited large number, and added loop-limit verification at end.
Please see the new commit.